### PR TITLE
[#69] feat(library): Custom runner type from string

### LINKED ISSUE #69: [Core feature request] Custom runner type from string
**What feature do you need?**
Runners are already supported by the library: https://github.com/krzema12/github-actions-kotlin-dsl/blob/6e7119ed9b55a9fd096b32046fcfa26fd93ad103/library/src/main/kotlin/it/krzeminski/githubactions/domain/RunnerType.kt#L7-L25

but it's not possible to define a custom one in runtime. Use cases:
* the library lags with adding a new type
* runner is passed from e.g. strategy matrix as a string, then it's not possible to use it

**Proposed implementation**  
Instead of `enum`, make it a `sealed class`, with additional
```kotlin
class Custom(val value) : RunnerType
```

### PR DESCRIPTION
**Pull request by jmfayard**
Resolves #69 

Changes `RunnerType` from an `Enum` to a `sealed class` with an additional `Custom("my-string")` possible value.

This allows do create a runner type from yet-unsupported value and also do use expressions like in

```kotlin
job(runsOn = RunnerType.Custom(expr("github.event.inputs.run-on")))
```

For the user, this changes how the imports work by default

`runsOn = UbuntuLatest`
to
`runsOn = RunnerType.UbuntuLatest`

### CODE REVIEW

File: library/src/main/kotlin/it/krzeminski/githubactions/yaml/JobsToYaml.kt [hunk 1]
```diff
@@ -68,8 +68,10 @@ private fun Job.toYaml() = buildString {
     append(customArgumentsToYaml())
 }
 
+/* Keep in sync with variable "allRunnerTypes" from "script-generator */
```
Comment (krzema12) [inline library/src/main/kotlin/it/krzeminski/githubactions/yaml/JobsToYaml.kt:71]: Maybe you could get all subclasses of `RunnerType` by reflection in the script-generator?

File: library/src/main/kotlin/it/krzeminski/githubactions/yaml/JobsToYaml.kt [hunk 2]
```diff
@@ -68,8 +68,10 @@ private fun Job.toYaml() = buildString {
...
 fun RunnerType.toYaml() =
     when (this) {
+        is RunnerType.Custom -> runsOn
```
Comment (krzema12) [inline library/src/main/kotlin/it/krzeminski/githubactions/yaml/JobsToYaml.kt:74]: Could you add a unit test for it?

also, nit:

```suggestion
        is Custom -> runsOn
```

File: library/src/main/kotlin/it/krzeminski/githubactions/domain/RunnerType.kt [hunk 3]
```diff
@@ -4,22 +4,25 @@ package it.krzeminski.githubactions.domain
  * Types of GitHub-hosted runners available. Should be kept in sync with the official list at
  * https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#choosing-github-hosted-runners
  */
-enum class RunnerType {
+sealed class RunnerType {
```
Comment (krzema12) [inline library/src/main/kotlin/it/krzeminski/githubactions/domain/RunnerType.kt:7]: nit: could be an interface - will remove the noise of writing `()` at the end of `RunnerType` for each type :smile: 

```suggestion
sealed interface RunnerType {
```

**Review by krzema12 — CHANGES_REQUESTED** [review]

File: library/src/main/kotlin/it/krzeminski/githubactions/yaml/JobsToYaml.kt (hunk 1 above)
Comment (jmfayard) [reply library/src/main/kotlin/it/krzeminski/githubactions/yaml/JobsToYaml.kt:71]: Done, that works

File: library/src/main/kotlin/it/krzeminski/githubactions/yaml/JobsToYaml.kt (hunk 2 above)
Comment (jmfayard) [reply library/src/main/kotlin/it/krzeminski/githubactions/yaml/JobsToYaml.kt]: done

File: library/src/main/kotlin/it/krzeminski/githubactions/domain/RunnerType.kt (hunk 3 above)
Comment (jmfayard) [reply library/src/main/kotlin/it/krzeminski/githubactions/domain/RunnerType.kt]: Nice

**Comment by krzema12** [comment]
Thanks! :100:

**Pull request merged**